python/src/com/jetbrains/python/testing/pyTestFixtures/PyTestFixtureReferenceContributor.kt

```diff
@@ -33,6 +35,45 @@ class PyTestFixtureReference(pyElement: PsiElement, fixture: PyTestFixture, priv
     return resultList.toArray(emptyArray())
   }
 
+  override fun getVariants(): Array<Any> {
+    // Provide completion variants for fixtures, especially inside pytest.mark.usefixtures strings
+    val element = myElement
+    val project = element.project
+    val context = TypeEvalContext.codeAnalysis(project, element.containingFile)
+
+    // If we are in a parameter, variants are handled elsewhere; be conservative and return empty
+    if (element !is PyStringLiteralExpression) return emptyArray()
+
+    val module = ModuleUtilCore.findModuleForPsiElement(element) ?: return emptyArray()
+
+    // Try to find the function we are attached to (decorated function)
+    val func = PsiTreeUtil.getParentOfType(element, PyFunction::class.java)
+
+    val fixtureNames: List<String> = if (func != null) {
+      // Use existing API that already filters fixtures applicable for a function and class context
+      getFixtures(module, func, context)
+        .filterNot { fixture ->
+          val fn = fixture.function ?: return@filterNot false
+          isFromPytestPackage(fn)
+        }
+        .map { it.name }
+    }
+    else {
+      // Fallback: collect all top-level fixtures available in module scope
+      findDecoratorsByName(module, TEST_FIXTURE_DECORATOR_NAMES)
```

Comment:
is there any reliable setup for this? as the test you added doesn't reach it

Reply:
I pushed two new tests for that branch and clarified the comment. It's for class-level and module-level usage of `pytest.mark.usefixtures`

Reply:
great, thank you